did-sdk/README.md

```diff
@@ -0,0 +1,49 @@
+# @adx/did-sdk
+
+An SDK for working with DIDs in ADX. Supports did:key, did:web, and did:ion.
+
+```typescript
+import crypto from 'crypto'
+import * as didSdk from '@adx/did-sdk'
+
+// resolve a did
+const did = await didSdk.resolve('did:web:example.com')
+console.log(did.getURI()) // => 'did:web:example.com'
+console.log(did.getController()) // => string
+console.log(did.listPublicKeys()) // => VerificationMethod[]
+console.log(did.getPublicKey('assertionMethod')) // => VerificationMethod
+console.log(did.listServices()) // => ServiceEndpoint[]
+console.log(did.getService('SomeService')) // => ServiceEndpoint
+
+// create a did
+const didKey = await didSdk.key.create('ed25519', {
+  secureRandom: () => crypto.randomBytes(32)
+})
+const didIon = await didSdk.ion.create(
+  {
+    services: [{
+      id: Buffer.from('#service1', 'utf8').toString('base64'),
```

Comment:
Can this id be anything? or is it convention that it's the base64 version of the service name? if so, maybe we move that into the library code?

Reply:
This is a really bizarre requirement that's enforced by the did-ion code and I have to see if it's a bug in their code or if the service IDs really do need to be base64 encoded.

Reply:
huh strange 🤔 

might make sense to move it into the library code then so that a developer doesn't have to see the strangeness

Reply:
🤦 It was the `#` that was choking it up. That "base64" validation is basically making sure the developer (me!) isn't putting values into the ID which can't be used in the fragment section of a URL correctly. Fixed by removing the pound sign.